model/project_parser.go

```diff
@@ -644,40 +691,67 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 		return nil, errors.Wrapf(err, LoadProjectError)
 	}
 
-	// return intermediateProject even if we run into issues to show merge progress
-	for _, path := range intermediateProject.Include {
+	if len(intermediateProject.Include) > 0 {
 		if opts == nil {
 			err = errors.New("trying to open include files with empty options")
 			return nil, errors.Wrapf(err, LoadProjectError)
 		}
-		opts.UpdateForFile(path.FileName)
-
-		var yaml []byte
-		opts.Identifier = identifier
-		opts.RemotePath = path.FileName
-		grip.Debug(message.Fields{
-			"message":     "retrieving included YAML file",
-			"remote_path": opts.RemotePath,
-			"read_from":   opts.ReadFileFrom,
-			"module":      path.Module,
-		})
-		if path.Module != "" {
-			yaml, err = retrieveFileForModule(ctx, *opts, intermediateProject.Modules, path.Module)
-		} else {
-			yaml, err = retrieveFile(ctx, *opts)
+		wg := sync.WaitGroup{}
+		outputYAMLs := make(chan yamlTuple, len(intermediateProject.Include))
+		includesToProcess := make(chan Include, len(intermediateProject.Include))
+
+		for _, path := range intermediateProject.Include {
+			includesToProcess <- path
 		}
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: retrieving file '%s'", LoadProjectError, path.FileName)
+
+		// Be polite. Don't make more than 10 concurrent requests to GitHub.
+		const maxWorkers = 10
+		workers := util.Min(maxWorkers, len(intermediateProject.Include))
+		for i := 0; i < workers; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for include := range includesToProcess {
+					processIntermediateProjectIncludes(ctx, identifier, intermediateProject, include, outputYAMLs, opts)
+				}
+			}()
 		}
-		add, err := createIntermediateProject(yaml, opts.UnmarshalStrict)
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: loading file '%s'", LoadProjectError, path.FileName)
+
+		// This order is deliberate:
+		// 1. Close `includesToProcess` so that the workers stop `range`ing over it.
+		// 2. Wait for the workers, since sending on a `nil` `outputYAMLs` would panic.
+		// 3. Close `outputYAMLs` so that the later `range` statement over it will stop after it's is drained.
+		close(includesToProcess)
+		wg.Wait()
+		close(outputYAMLs)
+
+		yamlMap := map[string][]byte{}
+		catcher := grip.NewBasicCatcher()
+		for elem := range outputYAMLs {
+			catcher.Add(elem.err)
+			if elem.yaml != nil {
+				yamlMap[elem.name] = elem.yaml
+			}
 		}
-		err = intermediateProject.mergeMultipleParserProjects(add)
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: merging file '%s'", LoadProjectError, path.FileName)
+
+		if catcher.HasErrors() {
+			return intermediateProject, errors.Wrap(catcher.Resolve(), "getting includes")
+		}
+
+		// We promise to iterate over includes in the order they are defined.
+		for _, path := range intermediateProject.Include {
+			add, err := createIntermediateProject(yamlMap[path.FileName], opts.UnmarshalStrict)
```

Comment:
For a bit of extra precaution against future changes, comma-ok checking the `yamlMap[path.FileName]` and returning a programmatic error might be preferable to avoid potential nil YAMLs causing panics during unmarshalling.

Reply:
done